tools/build_executable.py

```diff
@@ -0,0 +1,105 @@
+    parser.add_argument("--shell_bin")
+    parser.add_argument("--conda_env", default="spear-env")
+    parser.add_argument("--config_mode", default="Shipping")
+    parser.add_argument("--num_parallel_jobs", type=int, default=1)
+    parser.add_argument("--output_dir", default=os.path.dirname(os.path.realpath(__file__)))
+    args = parser.parse_args()
+
+    repo_dir = os.path.join(os.path.realpath(args.temp_dir), "spear")
+
+    # remove any contents already present for a clean build
+    if os.path.exists(repo_dir):
+        shutil.rmtree(repo_dir)
+        os.makedirs(repo_dir)
+
+    if sys.platform == "win32":
+        cmd_separator = "&"
+        run_uat_script = os.path.join(args.unreal_engine_dir, "Engine", "Build", "BatchFiles", "RunUAT.bat")
+        target_platform = "Win64"
+    elif sys.platform == "darwin":
+        cmd_separator = ";"
+        run_uat_script = os.path.join(args.unreal_engine_dir, "Engine", "Build", "BatchFiles", "RunUAT.sh")
+        target_platform = "Mac"
+    elif sys.plaform == "linux":
+        cmd_separator = ";"
+        run_uat_script = os.path.join(args.unreal_engine_dir, "Engine", "Build", "BatchFiles", "RunUAT.sh")
+        target_platform = "Linux"
+    else:
+        assert False
+
+    # clone repo along with submodules  
+    cmd = ["git", "clone", "--recurse-submodules", "https://github.com/isl-org/spear", repo_dir]
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd)
+    assert cmd_result.returncode == 0
+
+    # change working directory
+    cwd = os.getcwd()
+    print(f"[SPEAR | build.py] Changing working directory to {os.path.join(repo_dir, 'tools')}")
+    os.chdir(os.path.join(repo_dir, "tools"))
+
+    if args.commit_id:
+        cmd = ["git", "reset", "--hard", args.commit_id]
+        print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+        cmd_result = subprocess.run(cmd)
+        assert cmd_result.returncode == 0
+
+    # build thirdparty libs
+    cmd = ["python", "build_third_party_libs.py", "--num_parallel_jobs", f"{args.num_parallel_jobs}"]
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd, shell=True, executable=args.shell_bin)
+    assert cmd_result.returncode == 0
+
+    # create symbolic links
+    cmd = ["conda", "activate", args.conda_env + cmd_separator, "python", "create_symbolic_links.py"]
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd, shell=True, executable=args.shell_bin)
+    assert cmd_result.returncode == 0
+
+    # generate config file
+    cmd = ["conda", "activate", args.conda_env + cmd_separator, "python", "generate_config.py"]
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd, shell=True, executable=args.shell_bin)
+    assert cmd_result.returncode == 0
+    
+    archive_dir = os.path.join(args.output_dir, f"SpearSim-{target_platform}-{args.config_mode}")
```

Comment:
place short descriptive comment here for consistency with the rest of the file

Reply:
Done